classes/class-p4-settings.php

```diff
@@ -0,0 +1,193 @@
+			add_filter( 'cmb2_render_category_select_taxonomy', [ $this, 'p4_render_category_dropdown' ], 10, 2 );
+		}
+
+		/**
+		 * Register our setting to WP.
+		 */
+		public function init() {
+			register_setting( $this->key, $this->key );
+		}
+
+		/**
+		 * Add menu options page.
+		 */
+		public function add_options_page() {
+			$this->options_page = add_options_page( $this->title, $this->title, 'manage_options', $this->key, [ $this, 'admin_page_display' ] );
+		}
+
+		/**
+		 * Render act page dropdown.
+		 */
+		public function p4_render_act_page_dropdown( $field_args, $value ) {
+			wp_dropdown_pages( [
+					'show_option_none' => __( 'Select Page', 'planet4-master-theme' ),
+					'hide_empty'       => 0,
+					'hierarchical'     => true,
+					'selected'         => $value ,
+					'name'             => 'select_act_page',
+				]
+			);
+		}
+
+		/**
+		 * Render explore page dropdown.
+		 */
+		public function p4_render_explore_page_dropdown( $field_args, $value ) {
+			wp_dropdown_pages( [
+					'show_option_none' => __( 'Select Page', 'planet4-master-theme' ),
+					'hide_empty'       => 0,
+					'hierarchical'     => true,
+					'selected'         => $value ,
+					'name'             => 'select_explore_page',
+				]
+			);
+		}
+
+		/**
+		 * Render category dropdown.
+		 */
+		public function p4_render_category_dropdown( $field_args, $value ) {
+			wp_dropdown_categories( [
+					'show_option_none' => __( 'Select Category', 'planet4-master-theme' ),
+					'hide_empty'       => 0,
+					'hierarchical'     => true,
+					'orderby'          => 'name',
+					'selected'         => $value ,
+					'name'             => 'select_category',
+				]
+			);
+		}
+
+		/**
+		 * Admin page markup. Mostly handled by CMB2.
+		 */
+		public function admin_page_display() {
+			?>
+			<div class="wrap <?php echo $this->key; ?>">
+				<h2><?php echo esc_html( get_admin_page_title() ); ?></h2>
+				<?php cmb2_metabox_form( $this->option_metabox(), $this->key ); ?>
+			</div>
+			<?php
+		}
+
+		/**
+		 * Defines the theme option metabox and field configuration.
+		 * @return array
+		 */
+		public function option_metabox() {
+			return [
+				'id'         => 'option_metabox',
+				'show_on'    => [
+					'key'      => 'options-page',
+					'value'    => [
+						$this->key,
+					],
+				],
+				'show_names' => true,
+				'fields'     => $this->fields,
+			];
+		}
+
+		/**
+		 * Public getter method for retrieving protected/private variables.
+		 * @param  string  $field Field to retrieve
+		 * @return mixed          Field value or exception is thrown
+		 */
+		public function __get( $field ) {
+			// Allowed fields to retrieve.
+			if ( in_array( $field, [ 'key', 'fields', 'title', 'options_page' ], true ) ) {
+				return $this->{$field};
+			}
+
+			if ( 'option_metabox' === $field ) {
+				return $this->option_metabox();
+			}
+
+			throw new Exception( 'Invalid property: ' . $field );
+		}
```

Comment:
@sagarsdeshmukh 
Do we really need to declare this magic method?
I do not see any checks prior to returning the protected/private properties, so I do not see why to allow that.
Also, option_metabox() method is already publicly accessible.